src/parser/uri-utils.ts

```diff
@@ -110,6 +110,10 @@ export function isDomainLabelChar(char: string): boolean {
     return char === '_' || isDomainLabelStartChar(char);
 }
 
+export function hasDirectionalChar(char: string) {
```

Comment:
Minor, but can we rename this to `isDirectionOverrideChar`, and add a jsdoc comment above it explaining why it's used? 

Can add a link into the jsdoc for #377 too inside the jsdoc to point developers to the original issue that this new function solves